e2e/restartnode.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/spdk/spdk-csi/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var (
+	rpcClient util.RPCClient
+)
+
+var _ = ginkgo.Describe("SPDKCSI-CompleteWorkflow", func() {
+	f := framework.NewDefaultFramework("spdkcsi")
+
+	ginkgo.Context("Complete PVC and Snapshot Workflow", func() {
+		ginkgo.It("Create PVC, Snapshot, and Clone with Node Restart", func() {
+			var storageNodeID string
+			testPodLabel := metav1.ListOptions{
+				LabelSelector: "app=spdkcsi-pvc",
+			}
+			persistData := []string{"Data that needs to be stored"}
+			persistDataPath := []string{"/spdkvol/test"}
+
+			ginkgo.By("creating PVC and writing data")
+			deployPVC()
+			deployTestPod()
+			defer deleteTestPod()
+
+			err := waitForTestPodReady(f.ClientSet, 3*time.Minute)
+			if err != nil {
+				ginkgo.Fail(err.Error())
+			}
+
+			writeDataToPod(f, &testPodLabel, persistData[0], persistDataPath[0])
+
+			ginkgo.By("retrieving storage-node-id from PVC annotations")
+			pvc, err := f.ClientSet.CoreV1().PersistentVolumeClaims("spdk-csi").Get(context.TODO(), "spdkcsi-pvc", metav1.GetOptions{})
+			if err != nil {
+				ginkgo.Fail(err.Error())
+			}
+			storageNodeID = pvc.Annotations["simplybk/host-id"]
```

Comment:
storageNodeID should be retrieve from the mgmt plane and update the `pvc.yaml` file with the retrieved value